backend/TripBooker.HotelService/EventConsumers/Public/NewHotelUpdateEventConsumer.cs

```diff
@@ -0,0 +1,91 @@
+﻿using MassTransit;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using System.Transactions;
+using TripBooker.Common;
+using TripBooker.Common.TourOperator.Contract;
+using TripBooker.HotelService.Model;
+using TripBooker.HotelService.Model.Events.Hotel;
+using TripBooker.HotelService.Repositories;
+
+namespace TripBooker.HotelService.EventConsumers.Public;
+
+internal class NewHotelUpdateEventConsumer : IConsumer<HotelUpdateContract>
+{
+    private readonly IHotelEventRepository _eventRepository;
+    private readonly IHotelOccupationViewRepository _occupationRepository;
+    private readonly ILogger<NewHotelReservationEventConsumer> _logger;
+
+    public NewHotelUpdateEventConsumer(
+        IHotelEventRepository hotelRepository,
+        IHotelOccupationViewRepository occupationRepository,
+        ILogger<NewHotelReservationEventConsumer> logger)
+    {
+        _eventRepository = hotelRepository;
+        _occupationRepository = occupationRepository;
+        _logger = logger;
+    }
+
+    public async Task Consume(ConsumeContext<HotelUpdateContract> context)
+    {
+        _logger.LogInformation($"Hotel Update Contract recieved (HotelId = {context.Message.HotelId}, number of days = {context.Message.Days.Count})");
+
+        bool transactionSuccesfull;
+        do
+        {
+            transactionSuccesfull = true;
+
+            var hotelOccupations = new List<HotelOccupationModel>();
+
+            foreach (var day in context.Message.Days)
+            {
+                var hotelDay = await _occupationRepository.GetByHotelIdAndDateAsync(context.Message.HotelId, day, context.CancellationToken);
+                if (hotelDay != null)
+                    hotelOccupations.Add(hotelDay);
+                else
+                    _logger.LogInformation($"Could not locate HotelOccupationModel for given values: HotelId = {context.Message.HotelId}, DayTime = {day}");
+            }
+
+            try
+            {
+                await ValidateHotelUpdateTransaction(context.Message, hotelOccupations, context.CancellationToken);
+            }
+            catch (DbUpdateException e)
+            {
+                if (e.GetBaseException() is PostgresException { SqlState: GlobalConstants.PostgresUniqueViolationCode })
+                {
+                    // repeat if there was version violation, so the db read and business logic
+                    // does not need to be inside transaction
+                    transactionSuccesfull = false;
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+        }
+        while (!transactionSuccesfull);
+    }
```

Comment:
Zastanawiam się, czy nie powinniśmy tutaj zrobić request/response poprzez rabbita, i zwracać jakoś informację o zmianie, żeby dało się potem wyświetlić sensownie, że zmiana ze starego stanu na nowy, bo to będzie potrzebne w kontekście wyświetlenia ostatnich 10 zmian.

Nie jest to kluczowe ze względu na działanie samego mechanizmu, więc na razie może tylko TODO zostać na to.